### PR TITLE
Make it joomla 4 compatible

### PR DESCRIPTION
**Pull request by laoneo**
Makes web links installable and working on the current Joomla 4 branch. This fixes only the PHP errors. The layouts are not touched as we are still working on a proper BC solution https://github.com/joomla/joomla-cms/pull/17190.

### CODE REVIEW

File: src/administrator/components/com_weblinks/tables/weblink.php [hunk 1]
```diff
@@ -38,8 +38,11 @@ public function __construct(&$db)
 		// Set the published column alias
 		$this->setColumnAlias('published', 'state');
 
-		JTableObserverTags::createObserver($this, array('typeAlias' => 'com_weblinks.weblink'));
-		JTableObserverContenthistory::createObserver($this, array('typeAlias' => 'com_weblinks.weblink'));
+		if (version_compare(JVERSION, '4.0', '<' ) == 1)
+		{
+			JTableObserverTags::createObserver($this, array('typeAlias' => 'com_weblinks.weblink'));
```
Comment (zero-24) [inline src/administrator/components/com_weblinks/tables/weblink.php]: @yvesh this looks wrong to me as this means there are no `JTableObserverTags` call in 4.0. But as i have no idea what this thing is please take a look into that. Thanks.

Comment (laoneo) [reply src/administrator/components/com_weblinks/tables/weblink.php]: Observers got replaced by framework events. George should know the details.

Comment (wilsonge) [reply src/administrator/components/com_weblinks/tables/weblink.php]: We totally removed these things and there now should be plugin events for this built in

**Comment by wilsonge** [comment]
We need to make sure that `parent::check` is called in `WeblinksTable` so that we can have that plugin event properly triggered in the parent method

**Comment by laoneo** [comment]
Added the parent check call. All should be fine now.

**Comment by astridx** [comment]
Can we close this PR as this branch is merged into https://github.com/joomla-extensions/weblinks/pull/384 ?

**Comment by laoneo** [comment]
I would like to have a not namespaced version for Joomla 4 which is working and a namespaced version for Joomla 3.9 for testing. So we need this one here.

**Comment by mastacheata** [comment]
How is this PR related to the 4.0-dev branch?

This PR seems to only perform the minimally neccessary changes for it to work with Joomla 4.x
The Branch on the other hand is in a really strange state of something in between 3.x and 4.x style.
All of the Administrator stuff is namespaced and using 4.x style, while the component/frontend part seems to be untouched and still stuck in 3.x

Yet neither of the two seems to have adjusted to moving the dispatcher into it's own namespace as well.

**Pull request merged**